src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc

```diff
@@ -0,0 +1,2125 @@
+  std::vector<MessageSharedPtr>::iterator message_iterator =
+      messages_list_for_commands_to_hmi_.begin();
+  std::vector<utils::SharedPtr<am::commands::Command> >::iterator
+      command_iterator = commands_list_to_hmi_.begin();
+
+  for (; command_iterator != commands_list_to_hmi_.end(); ++command_iterator) {
+    EXPECT_CALL(app_mngr_, SendMessageToHMI(_));
+    (*command_iterator)->Run();
+    EXPECT_EQ(
+        (*(*message_iterator))[am::strings::params][am::strings::protocol_type],
+        kHmiProtocolType_);
+    EXPECT_EQ((*(*message_iterator))[am::strings::params]
+                                    [am::strings::protocol_version],
+              kProtocolVersion_);
+    ++message_iterator;
+  }
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnAppRegisteredNotificationSendNotificationToHmi) {
+  int32_t event_id = hmi_apis::FunctionID::INVALID_ENUM;
+  MessageSharedPtr message = CreateMessage();
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnAppRegisteredNotification>(message);
+
+  EXPECT_CALL(app_mngr_, SendMessageToHMI(_));
+  EXPECT_CALL(app_mngr_, event_dispatcher());
+  EXPECT_CALL(mock_event_dispatcher_, raise_event(_))
+      .WillOnce(GetEventId(&event_id));
+  command->Run();
+  EXPECT_EQ(hmi_apis::FunctionID::BasicCommunication_OnAppRegistered, event_id);
+  EXPECT_EQ((*message)[am::strings::params][am::strings::protocol_type],
+            kHmiProtocolType_);
+  EXPECT_EQ((*message)[am::strings::params][am::strings::protocol_version],
+            kProtocolVersion_);
+  EXPECT_CALL(app_mngr_, SendMessageToHMI(_)).Times(0);
+  EXPECT_CALL(app_mngr_, event_dispatcher()).Times(0);
+  EXPECT_CALL(mock_event_dispatcher_, raise_event(_)).Times(0);
+}
+
+// notifications(SendNotificationToMobile)
+
+TEST_F(HMICommandsNotificationsTest, CommandsNotificationEventDispatcher) {
+  int32_t event_id = hmi_apis::FunctionID::INVALID_ENUM;
+  std::vector<hmi_apis::FunctionID::eType>::iterator id_iterator =
+      id_list_for_commands_event_dispatcher_.begin();
+  std::vector<utils::SharedPtr<am::commands::Command> >::iterator
+      command_iterator = commands_list_event_dispatcher_.begin();
+  for (; command_iterator != commands_list_event_dispatcher_.end();
+       ++command_iterator) {
+    EXPECT_CALL(app_mngr_, event_dispatcher());
+    EXPECT_CALL(mock_event_dispatcher_, raise_event(_))
+        .WillOnce(GetEventId(&event_id));
+    (*command_iterator)->Run();
+    EXPECT_EQ((*id_iterator), event_id);
+    EXPECT_CALL(app_mngr_, event_dispatcher()).Times(0);
+    EXPECT_CALL(mock_event_dispatcher_, raise_event(_)).Times(0);
+    ++id_iterator;
+  }
+}
+
+TEST_F(HMICommandsNotificationsTest,
+       OnAppUnregisteredNotificationEventDispatcher) {
+  int32_t event_id = hmi_apis::FunctionID::INVALID_ENUM;
+  MessageSharedPtr message = CreateMessage();
+  utils::SharedPtr<am::commands::Command> command =
+      CreateCommand<am::commands::OnAppUnregisteredNotification>(message);
```

Comment:
@okozlovlux, Does not would be more readable setup `am::commands` namespace in current file?